pkg/controller/registry/reconciler/helpers.go

```diff
@@ -0,0 +1,65 @@
+package reconciler
+
+import (
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/install"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/ownerutil"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func DesiredRegistryNetworkPolicy(catalogSource client.Object, matchLabels map[string]string) *networkingv1.NetworkPolicy {
+	np := &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      catalogSource.GetName(),
+			Namespace: catalogSource.GetNamespace(),
+			Labels: map[string]string{
+				install.OLMManagedLabelKey: install.OLMManagedLabelValue,
+				CatalogSourceLabelKey:      catalogSource.GetName(),
+			},
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress, networkingv1.PolicyTypeEgress},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: ptr.To(corev1.ProtocolTCP),
+							Port:     ptr.To(intstr.FromInt32(50051)),
+						},
+					},
+				},
+			},
+		},
+	}
+	ownerutil.AddOwner(np, catalogSource, false, false)
+	return np
+}
+
+func sanitizedDeepEqual(a client.Object, b client.Object) bool {
```

Comment:
Given MatchLabels matters here and OLM has tendency to add new fields to labels from time to time, it may cause issue. I recommend adding a check for MatchLabels to ensure OLM will update NetworkPolicy if MatchLabels is updated.

Reply:
Now think of it, Semantic.DeepEqual check is probably enough here.

Reply:
Thanks for the review @dinhxuanvu!

If I'm understanding your comment correctly, I think this function would handle the case you are talking about.

What is essentially being ignored here are all the fields that we expect to be defaulted or managed by KAS. Every other diff (like us changing a label) will be different between `a` and `b` and thus cause this function to return `false` (for "not equal", and thus "need to update")